src/main/java/com/mongodb/quickstart/AggregationFramework.java

```diff
@@ -0,0 +1,69 @@
+package com.mongodb.quickstart;
+
+import com.mongodb.client.*;
+import org.bson.Document;
+import org.bson.conversions.Bson;
+import org.bson.json.JsonWriterSettings;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.function.Consumer;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import static com.mongodb.client.model.Accumulators.push;
+import static com.mongodb.client.model.Accumulators.sum;
+import static com.mongodb.client.model.Aggregates.*;
+import static com.mongodb.client.model.Filters.eq;
+import static com.mongodb.client.model.Projections.*;
+import static com.mongodb.client.model.Sorts.descending;
+
+public class AggregationFramework {
+
+    public static void main(String[] args) {
+        Logger.getLogger("org.mongodb.driver").setLevel(Level.WARNING);
+        String connectionString = System.getProperty("mongodb.uri");
+        try (MongoClient mongoClient = MongoClients.create(connectionString)) {
+            MongoDatabase db = mongoClient.getDatabase("sample_training");
+            MongoCollection<Document> zips = db.getCollection("zips");
+            MongoCollection<Document> posts = db.getCollection("posts");
+            threeMostPopulatedCitiesInTexas(zips);
+            threeMostPopularTags(posts);
+        }
+    }
+
+    /**
+     * find the 3 most densely populated cities in Texas.
+     * @param zips sample_training.zips collection from the MongoDB Sample Dataset in MongoDB Atlas.
+     */
+    private static void threeMostPopulatedCitiesInTexas(MongoCollection<Document> zips) {
+        Bson match = match(eq("state", "TX"));
+        Bson group = group("$city", sum("totalPop", "$pop"));
+        Bson project = project(fields(excludeId(), include("totalPop"), computed("city", "$_id")));
+        Bson sort = sort(descending("totalPop"));
+
+        ArrayList<Document> results = zips.aggregate(Arrays.asList(match, group, project, sort, limit(3)))
```

Comment:
Use `List<Document>` as the variable type.  No need to mention the concrete type twice.

Reply:
Automatic IntelliJ "store in variable feature"... Maybe some day they will get it right and use a List ^^.